### ISSUE TITLE
mypy returns strange error: Module has no attribute "TEMP_DIR"

### ISSUE THREAD

**Issue by jcrmatos** [python/mypy#6102]
I'm using Python 3.7.1 32b and mypy 0.650 on a Windows 7 Pro SP1.

I have 2 modules in the same directory.
Module base.py
...
`TEMP_DIR = APP_DIR + sep + 'temp' + sep  # type: str`

Module base_main.py
`import base`
...
`RECEIVER_STOP_PN = base.TEMP_DIR + 'STOP_RECEIVER'  # type: str`
...

mypy is returning the error Module has no attribute "TEMP_DIR"
on the RECEIVER_STOP_PN line of the base_main.py.

If I run the program which uses both files (base.py and base_main.py) everything works.
I use Flake8, Pylint, Bandit and Mypy.
Only Mypy returns an error.

I use the following command line
`mypy --cache-dir %TEMP%\.mypy_cache --strict-optional --warn-redundant-casts --disallow-untyped-defs --warn-unused-ignores --ignore-missing-imports --follow-imports silent --warn-no-return <filename_to_check>
`
Thanks

**Comment by gvanrossum** [comment]
Happy holidays! I suspect this has to do with eiher mypy's module search path or the presence of `__init__.py`, or both. You might get better diagnostics if you remove `--ignore-missing-imports --follow-imports` from your command line.

**Comment by jcrmatos** [comment]
Hello,

Happy holidays.

I have MYPYPATH and PYTHONPATH set to a directory with shared modules.
There is no `__init__.py` in my local directory. The path in MYPYPATH has an `__init__.py` but I tried removing it and errors are the same.

I removed the 2 arguments you mentioned, but it is even worse. Mypy shows an enormous list of errors, similar to these (where the TEMP_DIR error is included):

```
base_main.py:42: error: Module has no attribute "TEMP_DIR"
gui_change_config.py:43: error: No library stub file for standard library module
 'tkinter.scrolledtext'
gui_change_config.py:43: note: (Stub files are from https://github.com/python/ty
peshed)
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:188: error: Can
not find module named 'win32api'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:188: note: See
https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:192: error: Can
not find module named 'win32com.client'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:193: error: Can
not find module named 'win32com.shell'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:194: error: Can
not find module named 'win32com.shell.shell'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:195: error: Can
not find module named 'win32con'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:198: error: Can
not find module named 'win32console'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:199: error: Can
not find module named 'win32event'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:200: error: Can
not find module named 'win32file'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:201: error: Can
not find module named 'win32gui'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:203: error: Can
not find module named 'win32pdhutil'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:205: error: Can
not find module named 'win32process'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:210: error: Can
not find module named 'win32security'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:215: error: Can
not find module named 'win32service'
C:\Users\JMatos\OneDrive\Python\my_modules\my_modules\windows.py:219: error: Can
not find module named 'win32serviceutil'
```

All those errors from win32* happen because those "modules" are provided by pywin32.
Is there a way to solve them?

The 
"gui_change_config.py:43: error: No library stub file for standard library module
 'tkinter.scrolledtext'
gui_change_config.py:43: note: (Stub files are from https://github.com/python/ty
peshed)"
must be a typeshed thing, I presume.

Thanks,

JM

**Comment by jcrmatos** [comment]
Hello,

Happy holidays.

I have MYPYPATH and PYTHONPATH set to a directory with shared modules.
There is no `__init__.py` in my local directory. The path in MYPYPATH has an `__init__.py` but I tried removing it and errors are the same.

I removed the 2 arguments you mentioned, but the error still appears.

Thanks,

JM

**Comment by gvanrossum** [comment]
Hm, that `__init__.py` in MYPYPATH feels suspect. The reason is that when mypy finds a module in a folder with an `__init__.py` file it will assume that that folder is a package, and look up in its parent folder (or even further up if it finds an `__init__.py` there too).

If you want to debug this further I recommend creating a clean example in a separate directory containing just the code from your example -- see if you can repro the problem there.

**Comment by jcrmatos** [comment]
Hello,

With your suggestion I found the problem.
When I set MYPYPATH, I assumed that mypy would first look in my local directory and only if not found it would look into MYPYPATH.
That is not the case.
mypy looks first at MYPYPATH and it found a base.py in the shared modules directory that doesn't have the TEMP_DIR.
I changed the MYPYPATH to include the local directory first and now the errors are gone.

Thanks,

JM
